sessionctx/variable/session_test.go

```diff
@@ -57,91 +53,94 @@ func (*testSessionSuite) TestSetSystemVariable(c *C) {
 		{variable.TiDBMemQuotaApplyCache, "1024", false},
 		{variable.TiDBEnableStmtSummary, "1", false},
 	}
-	for _, t := range tests {
-		err := variable.SetSessionSystemVar(v, t.key, t.value)
-		if t.err {
-			c.Assert(err, NotNil)
-		} else {
-			c.Assert(err, IsNil)
-		}
+	for _, tc := range testCases {
+		t.Run(tc.key, func(t *testing.T) {
+			t.Parallel()
```

Comment:
this parallel may cause `fatal error: concurrent map writes`, because `variable.SetSessionSystemVar` will write variable to `SessionVars.systems`.   PTAL @penthaapatel @tisonkun @morgo

Reply:
Is there an issue logging this problem?

Also I don't find "SessionVars.systems" among the code base.

Reply:
I think you means we should run these subtests in serial because they share the state `v`, right?

Reply:
the failed ci https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/32947/pipeline

`SessionVars` is a struct and `systems` is its field

Reply:
> I think you means we should run these subtests in serial because they share the state v, right?

exactly, more specifically, `v.systems`

Reply:
Thanks! File an issue #27979 and a follow up to fix #27981